src/System.IO.FileSystem/src/System/IO/FileSystem.Unix.cs

```diff
@@ -12,6 +12,32 @@ internal static partial class FileSystem
     {
         internal const int DefaultBufferSize = 4096;
 
+        private static bool CopyDanglingSymlink(string sourceFullPath, string destFullPath)
+        {
+            // Check if the source is a dangling symlink. In those cases, we just want to copy the link
+            Interop.Sys.FileStatus ignored;
+            if (! (Interop.Sys.Stat(sourceFullPath, out ignored) < 0 &&
+                Interop.Sys.LStat(sourceFullPath, out ignored) == 0))
+            {
+                return false;
+            }
+
+            Interop.ErrorInfo errorInfo;
+            // get the target of the symlink
+            string linkTarget = Interop.Sys.ReadLink(sourceFullPath);
+            if (linkTarget is null)
+            {
+                errorInfo = Interop.Sys.GetLastErrorInfo();
+                throw Interop.GetExceptionForIoErrno(errorInfo, sourceFullPath);
+            }
+
+            if (Interop.Sys.Symlink(linkTarget, destFullPath) == 0)
+                return true;
+
+            errorInfo = Interop.Sys.GetLastErrorInfo();
+            throw Interop.GetExceptionForIoErrno(errorInfo, destFullPath);
```

Comment:
Nit: this can be:
```C#
// get the target of the symlink and create the link at the target location
string linkTarget = Interop.Sys.ReadLink(sourceFullPath);
if (linkTarget is null || Interop.Sys.Symlink(linkTarget, destFullPath) != 0)
{
    Interop.ErrorInfo errorInfo = Interop.Sys.GetLastErrorInfo();
    throw Interop.GetExceptionForIoErrno(errorInfo, sourceFullPath);
}

return true;
```